### PR TITLE
Add “nag” to call settings view that prods the user to change their p…

### PR DESCRIPTION
**Pull request by charlesmchen**
…rivacy settings after calls.

PTAL @michaelkirk 

### iPhone SE (names & numbers)

screen shot 2017-02-27 at 2 34 48 pm

### iPhone 7 Plus (names & numbers)

screen shot 2017-02-27 at 2 34 05 pm

### iPhone 7 Plus (CallKit + names & numbers)

screen shot 2017-02-27 at 1 52 58 pm

### iPhone SE (CallKit + names & numbers)

screen shot 2017-02-27 at 1 44 23 pm

### CODE REVIEW

**Comment by michaelkirk** [comment]
When I run this on the simulator and tapped "Show Privacy Settings", the app crashed, and saw this in the console:

```
2017-02-27 15:22:59:205 Signal[53374:2135488] [CallViewController] called didPressShowCallSettings(sender:)
2017-02-27 15:22:59:205 Signal[53374:2135488] [CallViewController] called markSettingsNagAsComplete()
2017-02-27 15:22:59.252 Signal[53374:2135488] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Storyboard (<UIStoryboard: 0x600000462f80>) doesn't contain a view controller with identifier 'SettingsNavigationController''
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010d57a34b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x000000010cd8321e objc_exception_throw + 48
	2   UIKit                               0x000000010a310e6f -[UIStoryboard instantiateInitialViewController] + 0
	3   Signal                              0x0000000105524e51 _TFFC6Signal18CallViewController24didPressShowCallSettingsFT6senderCSo8UIButton_T_U_FT_T_ + 1713
	4   Signal                              0x00000001054f3341 _TTRXFo___XFo_iT__iT__ + 17
	5   Signal                              0x0000000105525a81 _TPA__TTRXFo___XFo_iT__iT__ + 81
	6   Signal                              0x00000001054efaa0 _TTRXFo_iT__iT__XFo___ + 32
	7   Signal                              0x00000001054efad7 _TTRXFo___XFdCb___ + 39
	8   UIKit                               0x0000000109cbb684 -[UIPresentationController transitionDidFinish:] + 1289
	9   UIKit                               0x0000000109ed2f43 -[_UICurrentContextPresentationController transitionDidFinish:] + 42
	10  UIKit                               0x0000000109cbf120 __56-[UIPresentationController runTransitionForCurrentState]_block_invoke_2 + 183
	11  UIKit                               0x000000010a683fbd -[_UIViewControllerTransitionContext completeTransition:] + 102
	12  Foundation                          0x000000010861beb8 __NSFireDelayedPerform + 417
	13  CoreFoundation                      0x000000010d50c964 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 20
	14  CoreFoundation                      0x000000010d50c5f3 __CFRunLoopDoTimer + 1075
	15  CoreFoundation                      0x000000010d50c17a __CFRunLoopDoTimers + 250
	16  CoreFoundation                      0x000000010d503f01 __CFRunLoopRun + 2065
	17  CoreFoundation                      0x000000010d503494 CFRunLoopRunSpecific + 420
	18  GraphicsServices                    0x000000010f54ea6f GSEventRunModal + 161
	19  UIKit                               0x0000000109b3e964 UIApplicationMain + 159
	20  Signal                              0x00000001053b3577 main + 119
	21  libdyld.dylib                       0x000000010e7b568d start + 1
	22  ???                                 0x0000000000000003 0x0 + 3
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

File: Signal/src/view controllers/CallViewController.swift [hunk 1]
```diff
@@ -401,6 +473,11 @@ class CallViewController: UIViewController, CallObserver, CallServiceObserver, R
             incomingCallView.autoPinEdge(toSuperviewEdge:.bottom, withInset:incomingBottomMargin)
             incomingCallView.autoPinWidthToSuperview(withMargin:incomingHMargin)
             incomingCallView.setContentHuggingVerticalHigh()
+
+            // Settings nag views
+            settingsNagView.autoPinEdge(toSuperviewEdge:.bottom, withInset:settingsNagBottomMargin)
+            settingsNagView.autoPinWidthToSuperview(withMargin:settingsNagHMargin)
+            settingsNagView.autoPinEdge(.top, to:.bottom, of:callStatusLabel, withOffset:+avatarTopSpacing)
```
Comment (michaelkirk) [inline Signal/src/view controllers/CallViewController.swift]: what's that `+` in `+avatarTopSpacing` about?

File: Signal/src/view controllers/CallViewController.swift [hunk 2]
```diff
@@ -775,6 +904,56 @@ class CallViewController: UIViewController, CallObserver, CallServiceObserver, R
...
+                // Show nag for N seconds.
+                DispatchQueue.main.asyncAfter(deadline: .now() + 5) { [weak self] in
+                    guard let strongSelf = self else { return }
+                    strongSelf.dismissIfPossible(shouldDelay: false, ignoreNag: true)
+                }
+            }
+        } else if shouldDelay {
+            hasDismissed = true
+            DispatchQueue.main.asyncAfter(deadline: .now() + 1.5) {
+                self.dismiss(animated: true, completion:completion)
```
Comment (michaelkirk) [inline Signal/src/view controllers/CallViewController.swift]: Do you want this to be a weak reference, as above?

Comment (charlesmchen) [inline Signal/src/view controllers/CallViewController.swift]: Can't hurt. Done.

File: Signal/src/view controllers/CallViewController.swift (hunk 1 above)
Comment (charlesmchen) [inline Signal/src/view controllers/CallViewController.swift]: Removed.

**Comment by charlesmchen** [comment]
> When I run this on the simulator and tapped "Show Privacy Settings", the app crashed, and I saw this in the console:

Good catch. I habitually revert XCode's automated changes to storyboard files and I accidentally reverted a real change.

**Pull request merged**